File: bot/src/data/Subscription.ts
import { Sort } from "./Sort";

export interface Subscription {
    id: string,
    lastPostId: string,
    sort: Sort,
    subredditName: string,
    channels: ChannelSubscription[]
};

export interface ChannelSubscription {
    id: string,
    discordChannelId: string,
    guildId: string,
    isEnabled: boolean,
    isShowNsfw: boolean
};

File: api/src/Domain/Entities/Subscription.cs
using Domain.Entities.Common;
using Domain.Enums;

namespace Domain.Entities
{
    public class Subscription : BaseEntity
    {
        #region Properties

        public string LastPostId { get; set; } = "";

        public Sort Sort { get; set; }

        public Subreddit Subreddit { get; set; }

        public Guid SubredditId { get; set; }

        public ICollection<ChannelSubscription> ChannelSubscriptions { get; set; }

        #endregion

        #region Constructor

        public Subscription()
        {
            ChannelSubscriptions = new HashSet<ChannelSubscription>();
        }

        #endregion
    }
}
